rust/processor/src/bq_analytics/generic_parquet_processor.rs

```diff
@@ -140,25 +137,33 @@ where
             self.buffer.push(parquet_struct);
 
             if self.buffer_size_bytes >= self.max_buffer_size {
-                info!("Max buffer size reached, uploading to GCS.");
+                info!(
+                    table_name = ParquetType::TABLE_NAME,
+                    "Max buffer size reached, uploading to GCS."
+                );
                 if let Err(e) = self.upload_buffer(gcs_client).await {
                     error!("Failed to upload buffer: {}", e);
                     return Err(e);
                 }
                 self.last_upload_time = Instant::now();
             }
+        }
 
-            if self.last_upload_time.elapsed() >= self.upload_interval {
```

Comment:
this wasn't working for structs other than transactions when there aren't no structs to process for the batch we are processing. so it never triggers an upload even though it has past the interval duration provided in the config. so moving this out of the for loop and verified it's working